com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/StepsUtil.java

```diff
@@ -1080,6 +1082,17 @@ public static void setServicesGuids(DelegateExecution context, Map<String, Strin
         return JsonUtil.fromJson(jsonString, new TypeToken<Map<String, String>>() {
         }.getType());
     }
+    
+    public static CloudApplication getBoundApplication(List<CloudApplication> applications, UUID appGuid) {
+        return applications.stream()
+            .filter(app -> {
+                return app.getMeta()

```

Comment:
Extract the predicate in the filter to a separate method (hasGuid or something of the sort).